client.go

```diff
@@ -20,4 +20,5 @@ type Client interface {
 
 	Search(searchRequest *SearchRequest) (*SearchResult, error)
 	SearchWithPaging(searchRequest *SearchRequest, pagingSize uint32) (*SearchResult, error)
+	SearchAllowingPaging(searchRequest *SearchRequest) (*SearchResult, error)
```

Comment:
a name like SearchAllPages() might be clearer about what it is doing to the request